src/keys/x448.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+// https://tools.ietf.org/html/rfc7748
+
+use core::convert::TryInto;
+
+pub const PUBLIC_KEY_LEN: usize = 56;
+pub const SECRET_KEY_LEN: usize = 56;
+
+/// An X448 Shared Secret - the result of a Diffie-Hellman key exchange.
+///
+/// Each party computes this from an (ephemeral) [`SecretKey`] and their counterparty's [`PublicKey`].
+pub type SharedSecret = x448_crate::SharedSecret;
+
+/// An X448 Public Key
+pub struct PublicKey(x448_crate::PublicKey);
+
+impl PublicKey {
+    /// Load a [`PublicKey`] from a slice of bytes.
+    pub fn from_bytes(bytes: &[u8]) -> crate::Result<Self> {
```

Comment:
There should exist some crypto.rs-wide style-guide on how to implement such conversions: to/from array/slice own function vs trait method which can or can't fail. I'd propose the following rules:

- use standard traits as much as possible: AsRef, From/TryFrom, etc.
- use `_bytes` suffix if argument or return value is of type `[u8; N]` and use `_slice` suffix with the type is `&[u8]`
- use `try_` prefix with functions that can fail, ie. return `Result`